### ISSUE TITLE
If we do tests in a loop, it seems we don't have an easy way to prompt which iteration failed

### ISSUE THREAD

**Issue by catbro666** [jasmcaus/tau#38]
Do you have any advice for such user cases?

```c
int input[] = {1, 2, 3, 4};
int output[] = {1, 2, 3, 4};
for (int i = 0; i < 4, i++) {
  // ...
  CHECK_EQ(result, output[i]);
}
```

**Comment by jasmcaus** [comment]
Hmm, currently no. I suppose this is a very niche use case (to create an entirely separate macro). Are you tests working like this with this implementation?

**Comment by catbro666** [comment]
yeah, if all tests pass, everything is ok. If some tests fail, then I need to check them one by one to figure out which one fails.

**Comment by jasmcaus** [comment]
I see. I guess it makes sense to introduce a LOOP check then, but it'll be very restricted to just ints or max, floats. Interested in making a PR for this?

**Comment by catbro666** [comment]
OK, I can try.
